components/UserInfoCard.tsx

```diff
@@ -0,0 +1,324 @@
+import { Text, TextInput } from "./Themed";
+import {
+  Alert,
+  Image,
+  Modal,
+  StyleSheet,
+  View,
+  TouchableOpacity,
+} from "react-native";
+import React, { useEffect, useState } from "react";
+import { AntDesign, Feather } from "@expo/vector-icons";
+import {
+  deleteUser,
+  getAuth,
+  sendPasswordResetEmail,
+  signOut,
+} from "firebase/auth";
+import { auth } from "../firebase";
+import { useDispatch, useSelector } from "react-redux";
+import { currentReduxUser, setSignOutState } from "../redux/signin";
+import { useNavigation } from "@react-navigation/native";
+import { getAllDocsInCollection, getOneDocById } from "../helper";
+
+export const UserInfoCard = () => {
+  const [modalVisible, setModalVisible] = useState(false);
+  const [reviews, setReviews] = useState<string[]>([]);
+  const reviewsArray: any = [];
+  let reviewsArray2: string[] = [];
+  const user = useSelector(currentReduxUser);
```

Comment:
Same goes for this! First check if its current logged in user or some other users profile